docs/schemaform/walkthrough.md

```diff
@@ -0,0 +1,286 @@
+
+## JSON Schema
+
+One pre-requisite for understanding how schemaform works is the JSON Schema standard. JSON Schema is a way of describing the allowed shape of JSON objects. There are some good examples to look through on the (http://json-schema.org/examples.html)[JSON Schema site]. Here are some basics:
+
+Schema's have a type, that tells you what kind of data is allowed:
+
+```
+{ 
+  type: 'string'
+}
+```
+
+They can also have validation information, like regexes or length requirements:
+
+```
+{
+  type: 'string',
+  pattern: '^[ef]*$',
+  minLength: 2
+}
+```
+
+The above allows any string that's at least two characters and only contains `e` and `f`. So `eff` is valid, but `fcc` is not. You can also specify some built in `format` values for strings, like `email`, as a shortcut for including your own patterns.
+
+Objects fields can be described:
+
+```
+{
+  type: 'object',
+  properties: {
+    myField: {
+      type: 'number'
+    }
+  }
+}
+```
+
+This describes a json document that's an object with one property called `myField`, which is a number. So, `{ myField: 2 }` would be valid. 
+
+However, `{}` is also valid. If you want to required a property in an object, you use the `required` property:
+
+```
+{
+  type: 'object',
+  required: ['myField'],
+  properties: {
+    myField: {
+      type: 'number'
+    }
+  }
+}
+```
+
+Note that `required` is on the object that contains the field, not the field itself.
+
+Arrays work similarly to objects:
+
+```
+{
+  type: 'array',
+  items: {
+    type: 'boolean'
+  }
+}
+```
+
+This describes an array of boolean values: `[true, false, true]`. Items can be an object schema or any other type of schema as well.
+
+You can nest schemas as far down as you'd like. There are some other features, like metadata, sharing schema definitions between fields, and more complicated validation. But the above should get you most of the way there. There are many libraries that implement the JSON Schema spec and allow you to validate that an object matches a given schema. For reference, we use (https://www.npmjs.com/package/ajv)[ajv] and (https://www.npmjs.com/package/jsonschema)[jsonschema], the former in unit tests, and the latter in the schemaform code. ajv may go away eventually, since one of our dependencies is already using jsonschema.
+
+## How react-jsonschema-form works
+
+(https://github.com/mozilla-services/react-jsonschema-form)[react-jsonschema-form] (rjsf) generates a form from a JSON Schema, plus some other UI information. It does this by stepping through the schema depth first and rendering different components based on what type of data each property in the schema represents. You can try out the playground in the above link to get a feel for the resulting forms based on the schema inputs. We're going to look at how that library generates those forms in the rest of this section.
```

Comment:
maybe it's obvious but I would just specify: `rendering different *React* components` just to make clear that what it outputs is React components that come with the library (or custom ones).